test/nodes/test_label_generator.py

```diff
@@ -11,17 +12,44 @@
 @pytest.mark.parametrize("document_store", ["memory"], indirect=True)
 @pytest.mark.parametrize("retriever", ["embedding_sbert"], indirect=True)
 def test_pseudo_label_generator(
-    document_store, retriever: EmbeddingRetriever, question_generator: QuestionGenerator, tmp_path: Path
+    document_store: BaseDocumentStore,
+    retriever: EmbeddingRetriever,
+    question_generator: QuestionGenerator,
+    tmp_path: Path,
 ):
     document_store.write_documents(DOCS_WITH_EMBEDDINGS)
     psg = PseudoLabelGenerator(question_generator, retriever)
     train_examples = []
-    for idx, doc in enumerate(document_store):
-        output, stream = psg.run(documents=[doc])
-        assert "gpl_labels" in output
-        for item in output["gpl_labels"]:
-            assert "question" in item and "pos_doc" in item and "neg_doc" in item and "score" in item
-            train_examples.append(item)
+    output, pipe_id = psg.run(documents=document_store.get_all_documents())
+    assert "gpl_labels" in output
+    for item in output["gpl_labels"]:
+        assert "question" in item and "pos_doc" in item and "neg_doc" in item and "score" in item
+        train_examples.append(item)
+
+    assert len(train_examples) > 0
+    retriever.train(train_examples)
+    retriever.save(tmp_path)
```

Comment:
One more thought on this: do you think we really need to perform the training and save at each test? I think this is a very expensive operation that might slow down the suite a lot. I'd rather have a single dedicated test for it and remove it from all others, which can then count as unit tests.

Reply:
done @ZanSara also rebased to master.